templates/CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+# Contributing
+
+---
+
+COMMENT TO REMOVE
+
+Be aware of XXX in sentences and links: they need to be replaced by the name of the actual repository.
+
+---
+
+First of all, thank you for contributing to MeiliSearch! The goal of this document is to provide everything you need to know in order to contribute to MeiliSearch and its different integrations.
+
+<!-- MarkdownTOC autolink="true" style="ordered" indent="   " -->
+
+- [Assumptions](#assumptions)
+- [How to Contribute](#how-to-contribute)
+- [Development Workflow](#development-workflow)
+- [Git Guidelines](#git-guidelines)
+
+<!-- /MarkdownTOC -->
+
+## Assumptions
+
+1. **You're familiar with [GitHub](https://github.com) and the [Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)(PR) workflow.**
+2. **You've read the MeiliSearch [documentation](https://docs.meilisearch.com) and the [README](/README.md).**
+3. **You know about the [MeiliSearch community](https://docs.meilisearch.com/resources/contact.html). Please use this for help.**
+
+## How to Contribute
+
+1. Ensure your change has an issue! Find an [existing issue](https://github.com/meilisearch/XXX/issues/) or [open a new issue](https://github.com/meilisearch/XXX/issues/new). This is where you can get a feel if the change will be accepted or not.
+2. Once approved, [fork the XXX repository](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) in your own GitHub account.
```

Comment:
What's does it means exactly "approved"? Sorry, it isn't clear for me (so I guess to contributors that could be the case too). Maybe we can explain this a little bit, because there is nothing like an "intuitive" way of an issue getting "approved".

Reply:
I think @erlend-sh meant "the issue is approved to be implemented by a maintainer". It was clear for me, but maybe not for you @bidoubiwa @tpayet?

Reply:
Even if that seems clear, we never mentioned anywhere any process for 'reviewing' or 'approving' an issue. If we want that (I am not even sure that's necessary) we may be more explicit about it. Someone can for example fix an issue for himself/herself, then make a PR as a proposition (like in some 'this is my fix, take it if you want') and wont be waiting for an approval to make a fix. Is this the workflow we want?

Reply:
> we never mentioned anywhere any process for 'reviewing' or 'approving' an issue

What do you mean by anywhere?
If you mean in the SDK repositories, this file will be the actual place where the user can refer.
If you talk about our internal process, we are already working with issue approval (in integration-guides or during our meetings).

> Someone can for example fix an issue for himself/herself, then make a PR as a proposition

So, if I understand well, you disagree with the process "having an issue to do a PR", I mean the first point of the process: "Make sure that the contribution you want to make is explained or detailed in a GitHub issue! Find an [existing issue](https://github.com/meilisearch/XXX/issues/) or [open a new one](https://github.com/meilisearch/XXX/issues/new)."

I think the way "Issues->PR" is better than just "Nothing->PR" out of nowhere. At least in the process that we chose to publicly write down. But, of course, if a contributor does a PR directly, we are not going to ignore the PR.
**I just want our public and official process to be a common and clear process widely used in the OS community, even if everyone is not going to follow it** 🙂

What do you think @bidoubiwa @tpayet about having "Issue->PR" or "PR directly" in our official process?

Reply:
In official process Issue->PR since we do want to encourage good practices and visibility around the work that is being done and as you said it, everyone is not going to follow it :)

Reply:
> So, if I understand well, you disagree with the process "having an issue to do a PR"

You can totally do an Issue and a PR without having the approval for the issue. So no, this is not the same thing. I do not disagree with the idea of Requiring an Issue for a PR at all!

> What do you mean by anywhere?

I mean 'Anywhere' in the same guide. We are talking about an approval without saying what it is, where it comes from, or how you get it, or anything else about it, so I just prefer to make it clear

Reply:
Hum maybe I miss-understood your last comment 🤔 What you meant is to remove the "Once approved" part for an issue, right?

Again, it's our official process on the official place, so it's ok from my POV if we let the "Once approved": of course the user does not need to necessarily follow it. He/she indeed can suggest the PR if it's already done, not need to wait.
But if you think guys it's too confusing, I'm not against to remove it either.